File: python/rounding.py
# rounding.py
# This program rounds numbers the user enters.
# Github: https://www.github.com/0x4248/examples
# Licence: GNU General Public License v3.0
# By: 0x4248

# This program rounds numbers the user enters.
while True:
    # Ask the user to enter a number
    number = input("Enter a number: ")
    # Convert the number to a float
    number = float(number)
    # Get the number of decimal places
    decimal_places = len(str(number).split(".")[1])
    # Round the number to the number of decimal places
    for i in range(decimal_places):
        # Print the rounded number
        print("Round to", i, "decimal places:", round(number, i))

File: c/rounding.c
/**
 * @file rounding.c
 * @brief Rounding numbers in C
 * @details Turns a number form a float to an int, rounding up or down depending
 * on the decimal value.
 * @github: https://www.github.com/0x4248/examples
 * @license: GNU General Public License v3.0
 * @author Lewis Evans
 */

#include <math.h>
#include <stdio.h>

int main() {
    float a;
    int b;
    while (0 == 0) {
        printf("Enter a number: ");
        scanf("%f", &a);

        b = (int)a;
        if (a - b >= 0.5) {
            b++;
        }
        printf("Rounded: %d\n", b);
    }
    return 0;
}